Calliope App/Bluetooth/CalliopeTypes/FlashableCalliope.swift

```diff
@@ -255,14 +280,16 @@ class FlashableCalliope: CalliopeBLEDevice {
 
     private func sendNextPackages() {
         updateCallback("send 4 packages beginning at \(startPackageNumber)")
-        guard var partialFlashData = partialFlashData else {
+        /* guard var partialFlashData = partialFlashData else {
+        } */
+        if (partialFlashData == nil) {
```

Comment:
Wenn du diese Zeilen nicht geändert hättest bräuchtest du unten das "!" zum force-unwrap nicht.

Reply:
Danke für die Erklärung! Jetzt verstehe ich den Fehler. Ich habe nicht an die Value-Semantik von Structs in Swift gedacht. 

Trotzdem wäre hier die guard clause in Swift das richtige (guard partialFlashData != nil else...). Am elegantesten wäre vermutlich, die veränderten partialFlashData wieder in die globalen zurück zu schreiben.

Reply:
Ich habe es wieder zurück auf 'guard var' gesetzt und dann einfach die globale Kopie mit der Lokalen überschrieben, was erstaunlicher Weise einfach funktioniert hat.

Nebenbei habe ich noch den Zusatzparameter für den Dialog von 'partial' in 'partialFlashing' umbenannt.